develop/interoperability/xcm-config.md

```diff
@@ -0,0 +1,245 @@
+---
+title: XCM Config
+description: Learn how the XCM Executor configuration works for your custom Polkadot SDK-based runtime with detailed guidance and references.
+---
+
+# XCM Config
+
+## Introduction
+
+The [XCM executor](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/xcm/xcm-executor/src){target=\_blank} is a crucial component responsible for interpreting and executing XCM messages with Polkadot SDK-based chains. It processes and manages XCM instructions, ensuring they are executed correctly and in sequentially. Adhering to the Cross-Consensus Virtual Machine (XCVM) specification, the XCM executor can be customized or replaced with an alternative that also complies with the [XCVM standards](https://github.com/polkadot-fellows/xcm-format?tab=readme-ov-file#12-the-xcvm){target=\_blank}.
+
+The `XcmExecutor` is not a pallet but a struct parameterized by a `Config` trait. The `Config` trait is the inner configuration, parameterizing the outer `XcmExecutor<Config>` struct. Both configurations are set up within the runtime.
+
+The executor is highly configurable, with the [XCM builder](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/xcm/xcm-builder/src){target=\_blank} offering building blocks to tailor the configuration to specific needs. While they serve as a foundation, users can easily create custom blocks to suit unique configurations. Users can also create their building blocks to address unique needs. This article examines the XCM configuration process, explains each configurable item, and provides examples of the tools and types available to help customize these settings.
+
+## XCM Executor Configuration
+
+The `Config` trait defines the XCM executor’s configuration, which requires several associated types. Each type has specific trait bounds that the concrete implementation must fulfill. Some types, such as `RuntimeCall`, come with a default implementation in most cases, while others use the unit type `()` as the default. For many of these types, selecting the appropriate implementation carefully is crucial. Predefined solutions and building blocks can be adapted to your specific needs. These solutions can be found in the [`xcm-builder`](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/xcm/xcm-builder){target=\_blank} folder.
+
+Each type is explained below, along with an overview of some of its implementations:
+
+```rust
+--8<-- 'code/develop/interoperability/xcm-config/xcm-config-trait.rs'
+```
+
+## Config Items
```

Comment:
There are some terms like `Multilocation`, `RuntimeOrigin` and so on that are not linked to an explanation. We might need a exclusive glossary for XCM or add these terms to the existing one.

Reply:
Agreed, also, we should use just `Location` going forward, remove the `Multi` prefix altogether

Reply:
agree, moving this item to post-launch, thanks!